src/blocks/content-helper/content-helper-provider.ts

```diff
@@ -80,7 +83,8 @@ class ContentHelperProvider {
 			return Promise.reject( error );
 		}
 
-		let message = `${ __( 'Top-performing posts', 'wp-parsely' ) } ${ apiQuery.message }.`;
+		/* translators: %s: message such as "from category Foo", %d: number of days */
+		let message = sprintf( __( 'Top-performing posts %1$s in last %2$d days.', 'wp-parsely' ), apiQuery.message, RELATED_POSTS_DEFAULT_TIME_RANGE );
```

Comment:
In English, this string will be broken if the number of days is 1.

Use https://developer.wordpress.org/block-editor/reference-guides/packages/packages-i18n/#n to provide a singular and plural string, and let the i18n system use the correct [translated] string.

Reply:
Because of the extra string placeholder, it makes it a little more complex, so this may need to be of the form:

```js
let message = _n( 
	sprintf(
		__('Top-performing posts %1$s in last %2$d day.', 'wp-parsely' ),
		apiQuery.message,
		RELATED_POSTS_DEFAULT_TIME_RANGE
	),
	sprintf(
		__('Top-performing posts %1$s in last %2$d days.', 'wp-parsely' ),
		apiQuery.message,
		RELATED_POSTS_DEFAULT_TIME_RANGE
	),
	RELATED_POSTS_DEFAULT_TIME_RANGE
);
```

That is (working from the inside out), it creates a singular and plural translatable string, which after translation then has its placeholders replaced, and then the correct string is assigned to `message` depending on the value of the time range constant.

Reply:
Actually I was not making these changes because by default API returns data for last 3 days and have a plan to do these changes while adding API params filter (next ticket in which I will work).

BTW based on your feedback I did tried to do this but facing following error which I was unable to solve

![image](https://user-images.githubusercontent.com/31419912/204219645-f2b57a94-8972-46d7-a5b9-608953a8b465.png)

Please do let me know the proper way to solve this and then I will make the changes in the next PR.

Reply:
I don't think that we have to look into this in this PR, since returning data for 1 day is not possible. When/if returning data for 1 day becomes available, we can look into plural/singular options.